File: backend/app/models/tournament.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.db import Base


class Tournament(Base):
    __tablename__ = "tournaments"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)

    name: Mapped[str] = mapped_column(String(160), nullable=False)
    year: Mapped[int] = mapped_column(Integer, nullable=False, index=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)

    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=datetime.utcnow,
        nullable=False,
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        nullable=False,
    )

    matches = relationship("Match", back_populates="tournament")

File: frontend/src/types/tournament.ts
export type Tournament = {
  id: number;
  name: string;
  year: number;
  description?: string | null;
  is_active: boolean;
  created_at: string;
};

export type TournamentCreatePayload = {
  name: string;
  year: number;
  description?: string | null;
  is_active: boolean;
};

export type TournamentUpdatePayload = {
  name?: string;
  year?: number;
  description?: string | null;
  is_active?: boolean;
};
